ledger/sandbox/src/main/scala/com/digitalasset/platform/index/EventFilter.scala

```diff
@@ -35,6 +35,7 @@ object EventFilter {
 
   def apply(event: ActiveContract)(txf: TransactionFilter): Option[ActiveContract] =
     Some(event.copy(witnesses = event.witnesses.filter(included(_, event.contract.template, txf))))
-      .filter(_.witnesses.nonEmpty)
+      .filter(ac =>
+        (ac.signatories union ac.observers).exists(included(_, event.contract.template, txf)))
```

Comment:
It would probably be better to do this check upfront to not filter witnesses needlessly. I'm still not certain about the relationship between stakeholders and witnesses, but if there is a way to not run the event through the filter twice it would be preferable.

Reply:
Fixed in 1b0ccf3cf.